### PR TITLE
cloudcore support --defaultconfig subcommand

### PR DESCRIPTION
**Pull request by kadisi**
Signed-off-by: zhangjie <iamkadisi@163.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

 /kind feature

**What this PR does / why we need it**:
cloudcore support --defaultconfig subcommand:
```
# ./cloudcore --defaultconfig


# With --defaultconfig flag, users can easily get a default full config file as reference, with all fields (and field descriptions) included and default values set.
# Users can modify/create their own configs accordingly as reference.
# Because it is a full configuration, it is more suitable for advanced users.

apiVersion: cloudcore.config.kubeedge.io/v1alpha1
kind: CloudCore
kubeAPIConfig:
  burst: 200
  contentType: application/vnd.kubernetes.protobuf
  kubeConfig: /root/.kube/config
  master: ""
  qps: 100
modules:
  cloudhub:
    enable: true
    keepaliveInterval: 30
    nodeLimit: 10
    quic:
      address: 0.0.0.0
      maxIncomingStreams: 10000
      port: 10001
    tlsCAFile: /etc/kubeedge/ca/rootCA.crt
    tlsCertFile: /etc/kubeedge/certs/edge.crt
    tlsPrivateKeyFile: /etc/kubeedge/certs/edge.key
    unixsocket:
      address: unix:///var/lib/kubeedge/kubeedge.sock
      enable: true
    websocket:
      address: 0.0.0.0
      enable: true
      port: 10000
    writeTimeout: 30
  devicecontroller:
    buffer:
      deviceEvent: 1
      deviceModelEvent: 1
      updateDeviceStatus: 1024
    context:
      receiveModule: devicecontroller
      responseModule: cloudhub
      sendModule: cloudhub
    enable: true
    load:
      updateDeviceStatusWorkers: 1
  edgecontroller:
    buffer:
      configmapEvent: 1
      endpointsEvent: 1
      podEvent: 1
      queryConfigmap: 1024
      queryEndpoints: 1024
      queryNode: 1024
      queryPersistentvolume: 1024
      queryPersistentvolumeclaim: 1024
      querySecret: 1024
      queryService: 1024
      queryVolumeattachment: 1024
      secretEvent: 1
      serviceEvent: 1
      updateNode: 1024
      updateNodeStatus: 1024
      updatePodStatus: 1024
    context:
      receiveModule: edgecontroller
      responseModule: cloudhub
      sendModule: cloudhub
    enable: true
    load:
      queryConfigmapWorkers: 4
      queryEndpointsWorkers: 4
      queryNodeWorkers: 4
      queryPersistentColumeClaimWorkers: 4
      queryPersistentVolumeWorkers: 4
      querySecretWorkers: 4
      queryServiceWorkers: 4
      queryVolumeAttachmentWorkers: 4
      updateNodeStatusWorkers: 1
      updateNodeWorkers: 4
      updatePodStatusWorkers: 1
    nodeUpdateFrequency: 10


```
**Which issue(s) this PR fixes**:

Ref: #1171 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
cloudcore support --defaultconfig subcommand
```

### CODE REVIEW

**Comment by kevin-wangzefeng** [comment]
could you find a way to remove the duplicated functions? And it looks weird that minconfig functions and default functions mixed together.

**Comment by kadisi** [comment]
> could you find a way to remove the duplicated functions? And it looks weird that minconfig functions and default functions mixed together.

ok ,  i will change `type minConfigValue and  defaultConfigValue`  into one type

File: pkg/util/flag/flags.go [hunk 1]
```diff
@@ -12,18 +12,119 @@ distributed under the License is distributed on an "AS IS" BASIS,
...
+func AddFlags(fs *pflag.FlagSet) {
+	fs.AddFlag(pflag.Lookup(minConfigFlagName))
+	fs.AddFlag(pflag.Lookup(defaultConfigFlagName))
+}
+
+// PrintMinConfigAndExitIfRequested will check if the -minconfig flag was passed
+// and, if so, print the min config and exit.
+func PrintMinConfigAndExitIfRequested() {
+	if *minConfigFlag == ConfigTrue {
+		config := v1alpha1.NewMinCloudCoreConfig()
```
Comment (fisherxu) [inline pkg/util/flag/flags.go]: How about set the `v1alpha1.NewMinCloudCoreConfig()` as a parameter for PrintMinConfigAndExitIfRequested？

Comment (kadisi) [reply pkg/util/flag/flags.go]: sound good

**Pull request merged**